examples/simple_hpc_pbs_ldap/scripts/ldap_add_user.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+script_name=$(basename "$0")
+
+server_hostname=$HOSTNAME
+ldap_base_dn="DC=${server_hostname},DC=local"
+
+function display_help {
+    cat <<EOF
+$script_name 
+
+Add an LDAP user.
+
+Usage:
+  $script_name --username USERNAME --user-id USER_ID
+               [--ldap-admin LDAP_ADMIN]
+               [--ldap-password-file LDAP_PASSWORD_FILE]
+               [--sudo] [--ssh-key SSH_KEY] [--home-root HOME_DIR]
+               [--password PASSWORD] [--ssh-key SSH_KEY]
+
+Options:
+  -h, --help            Show this help message.
+  --ldap-admin LDAP_ADMIN
+                        The LDAP admin username
+                        [default: admin]
+  --ldap-password-file LDAP_PASSWORD_FILE
+                        A file containing the LDAP admin password
+                        [default: /root/ldap_admin_password.txt]
+  --home-root HOME_DIR  The root for home directories.
+                        [default: /share/home]
+  --username USERNAME   The users name to add
+  --password PASSWORD   The users password (empty means random string)
+                        [default: <empty-string>]
+  --user-id  USER_ID    The UID (also used for GID) to use
+  --ssh-key SSH_KEY     An additional public key to add to authorized_keys
+  --sudo                Give user sudo privilidges
+EOF
+}
+
+password=
+ldap_admin_username=admin
+ldap_admin_password_file=/root/ldap_admin_password.txt
+home_root=/share/home
+username=
+password=$(slappasswd -g)
+user_id=
+add_sudo=
+ssh_key=
+
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+
+    case $key in
+        -h|--help)
+            display_help
+            exit 0
+            ;;
+        --ldap-admin)
+            ldap_admin_username="$2"
+            shift 2
+            ;;
+        --ldap-password-file)
+            ldap_admin_password_file="$2"
+            shift 2
+            ;;
+        --home-root)
+            home_root="$2"
+            shift 2
+            ;;
+        --username)
+            username="$2"
+            shift 2
+            ;;
+        --password)
+            password="$2"
+            shift 2
+            ;;
+        --user-id)
+            user_id="$2"
+            shift 2
+            ;;
+        --ssh-key)
+            ssh_key="$2"
+            shift 2
+            ;;
+        --sudo)
+            add_sudo=yes
+            shift
+            ;;
+        *)    
+            # unknown option
+            echo "ERROR: unknown option - $1"
+            echo
+            display_help
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$username" -o -z "$user_id" ]; then
+    echo "ERROR: missing mandatory argument(s)"
+    echo
+    display_help
+    exit 1
+fi
+
+user_ldif="ldap_user_${username}.ldif"
+cat <<EOF >$user_ldif
+dn: uid=${username},ou=People,${ldap_base_dn}
+objectClass: top
+objectClass: person
+objectClass: posixAccount
+objectClass: shadowAccount
+objectClass: inetOrgPerson
+objectClass: organizationalPerson
+cn: $username
+sn: $username
```

Comment:
Not a big issue, but worth considering adhering with the RFC2253 standard:
```
cn: Davide Vanzo
sn: Vanzo
uid: vanzod
```

Reply:
Yes - I did realise this although do we want to make users enter real names for setting this up?